src/libraries/Microsoft.Extensions.Options/src/ConfigureNamedOptions.cs

```diff
@@ -16,7 +16,7 @@ public class ConfigureNamedOptions<TOptions> : IConfigureNamedOptions<TOptions>
         /// </summary>
         /// <param name="name">The name of the options.</param>
         /// <param name="action">The action to register.</param>
-        public ConfigureNamedOptions(string name, Action<TOptions> action)
+        public ConfigureNamedOptions(string? name, Action<TOptions>? action)
```

Comment:
I wonder if this makes sense. This is a `ConfigureNamedOptions` class for a "named options". Why would `null` be a valid name?

Even though it doesn't throw ArgNull, I wonder if this should be `string name`.

Reply:
It avoided needing to introduce a special value for ConfigureAll, because null name is generally illegal, so it was safe to use for https://github.com/dotnet/runtime/blob/22a71c8a1d4ff224e739d49eaf75597b5a89b8fb/src/libraries/Microsoft.Extensions.Options/src/OptionsServiceCollectionExtensions.cs#L67

Reply:
Ah this makes sense. Thanks!